File: Views/Onboarding/ConfidenceStepView.swift
import SwiftUI

struct ConfidenceStepView: View {
    @ObservedObject var viewModel: OnboardingViewModel
    
    private let confidenceOptions = [
        "confidence_completely",
        "confidence_most_of_time",
        "confidence_not_always"
    ]
    
    var body: some View {
        VStack(spacing: 0) {
            // Espace entre barre de progression et titre (harmonisé)
            Spacer()
                .frame(height: 40)
            
            // Titre centré à gauche
            HStack {
                Text("confidence_question".localizedOnboarding)
                    .font(.system(size: 28, weight: .bold))
                    .foregroundColor(.black)
                    .multilineTextAlignment(.leading)
                Spacer()
            }
            .padding(.horizontal, 30)
            
            // Sous-titre explicatif
            HStack {
                Text("private_answer_note".localizedOnboarding)
                    .font(.system(size: 14, weight: .regular))
                    .foregroundColor(.black.opacity(0.6))
                    .multilineTextAlignment(.leading)
                Spacer()
            }
            .padding(.horizontal, 30)
            .padding(.top, 8)
            
            Spacer()
            
            // Options de réponse
            VStack(spacing: 12) {
                ForEach(Array(confidenceOptions.enumerated()), id: \.offset) { index, localizationKey in
                    Button(action: {
                        print("🔥 ConfidenceStepView: Option sélectionnée: \(localizationKey)")
                        viewModel.confidenceAnswer = localizationKey
                    }) {
                        Text(localizationKey.localizedOnboarding)
                            .font(.system(size: 16))
                            .foregroundColor(viewModel.confidenceAnswer == localizationKey ? .white : .black)
                            .multilineTextAlignment(.leading)
                            .lineLimit(nil)
                            .fixedSize(horizontal: false, vertical: true)
                            .frame(maxWidth: .infinity, alignment: .leading)
                            .padding(.vertical, 16)
                            .padding(.horizontal, 20)
                            .background(
                                RoundedRectangle(cornerRadius: 12)
                                    .fill(viewModel.confidenceAnswer == localizationKey ? Color(hex: "#FD267A") : Color.white)
                                    .overlay(
                                        RoundedRectangle(cornerRadius: 12)
                                            .stroke(
                                                viewModel.confidenceAnswer == localizationKey ? Color(hex: "#FD267A") : Color.black.opacity(0.1),
                                                lineWidth: viewModel.confidenceAnswer == localizationKey ? 2 : 1
                                            )
                                    )
                                    .shadow(color: Color.black.opacity(0.05), radius: 8, x: 0, y: 4)
                            )
                    }
                    .buttonStyle(PlainButtonStyle())
                }
            }
            .padding(.horizontal, 30)
            
            Spacer()
            
            // Zone blanche collée en bas avec bouton "Continuer"
            VStack(spacing: 0) {
                Button(action: {
                    print("🔥 ConfidenceStepView: Bouton continuer pressé")
                    viewModel.nextStep()
                }) {
                    Text("continue".localized)
                        .font(.system(size: 18, weight: .semibold))
                        .foregroundColor(.white)
                        .frame(maxWidth: .infinity)
                        .padding(.vertical, 16)
                        .background(Color(hex: "#FD267A"))
                        .cornerRadius(28)
                        .opacity(viewModel.confidenceAnswer.isEmpty ? 0.5 : 1.0)
                }
                .disabled(viewModel.confidenceAnswer.isEmpty)
                .padding(.horizontal, 30)
            }
            .padding(.vertical, 30)
            .background(Color.white)
        }
        .background(Color(red: 0.97, green: 0.97, blue: 0.98))
    }
}

#Preview {
    ConfidenceStepView(viewModel: OnboardingViewModel())
}

File: Views/Onboarding/ConfidenceStepView.kt
package com.love2love.onboarding

import android.util.Log
import androidx.annotation.StringRes
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.alpha
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewmodel.compose.viewModel

// --- UI constants ---
private val AccentPink = Color(0xFFFD267A)
private val ScreenBg = Color(0xFFF7F7FA)
private val CardBorder = Color.Black.copy(alpha = 0.10f)

// --- Model for options ---
data class ConfidenceOption(
    val key: String,                 // ex: "confidence_completely"
    @StringRes val labelRes: Int     // ex: R.string.confidence_completely
)

// --- ViewModel équivalent ---
class OnboardingViewModel : ViewModel() {
    var confidenceAnswer by mutableStateOf("")   // garde la clé (ex: "confidence_completely")
        private set

    fun selectAnswer(key: String) {
        Log.d("🔥 ConfidenceStepView", "Option sélectionnée: $key")
        confidenceAnswer = key
    }

    fun nextStep() {
        Log.d("🔥 ConfidenceStepView", "Bouton continuer pressé avec réponse: $confidenceAnswer")
        // TODO: navigation/étape suivante
    }
}

@Composable
fun ConfidenceStepScreen(
    viewModel: OnboardingViewModel = viewModel()
) {
    val context = LocalContext.current

    // Mapping des options -> resources Android (labels dans strings.xml)
    val confidenceOptions = remember {
        listOf(
            ConfidenceOption("confidence_completely",  R.string.confidence_completely),
            ConfidenceOption("confidence_most_of_time", R.string.confidence_most_of_time),
            ConfidenceOption("confidence_not_always",   R.string.confidence_not_always)
        )
    }

    val canContinue = viewModel.confidenceAnswer.isNotEmpty()

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(ScreenBg)
    ) {
        // Espace entre barre de progression et titre (40)
        Spacer(modifier = Modifier.height(40.dp))

        // Titre
        Row(
            modifier = Modifier.padding(horizontal = 30.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            Text(
                text = context.getString(R.string.confidence_question),
                fontSize = 28.sp,
                fontWeight = FontWeight.Bold,
                color = Color.Black,
                textAlign = TextAlign.Start,
                modifier = Modifier.weight(1f)
            )
        }

        // Sous-titre
        Row(
            modifier = Modifier
                .padding(horizontal = 30.dp)
                .padding(top = 8.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            Text(
                text = context.getString(R.string.private_answer_note),
                fontSize = 14.sp,
                fontWeight = FontWeight.Normal,
                color = Color.Black.copy(alpha = 0.6f),
                textAlign = TextAlign.Start,
                modifier = Modifier.weight(1f)
            )
        }

        Spacer(modifier = Modifier.weight(1f))

        // Options
        Column(
            modifier = Modifier.padding(horizontal = 30.dp),
            verticalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            confidenceOptions.forEach { option ->
                val selected = viewModel.confidenceAnswer == option.key
                Surface(
                    shape = MaterialTheme.shapes.medium,
                    color = if (selected) AccentPink else Color.White,
                    tonalElevation = 0.dp,
                    shadowElevation = 8.dp,
                    border = BorderStroke(
                        width = if (selected) 2.dp else 1.dp,
                        color = if (selected) AccentPink else CardBorder
                    ),
                    modifier = Modifier
                        .fillMaxWidth()
                        .clickable { viewModel.selectAnswer(option.key) }
                ) {
                    Text(
                        text = context.getString(option.labelRes),
                        fontSize = 16.sp,
                        color = if (selected) Color.White else Color.Black,
                        lineHeight = 22.sp,
                        modifier = Modifier
                            .fillMaxWidth()
                            .padding(vertical = 16.dp, horizontal = 20.dp)
                    )
                }
            }
        }

        Spacer(modifier = Modifier.weight(1f))

        // Zone blanche en bas + bouton Continuer
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .background(Color.White)
                .padding(vertical = 30.dp)
        ) {
            Button(
                onClick = { viewModel.nextStep() },
                enabled = canContinue,
                shape = MaterialTheme.shapes.extraLarge,
                colors = ButtonDefaults.buttonColors(containerColor = AccentPink),
                modifier = Modifier
                    .padding(horizontal = 30.dp)
                    .fillMaxWidth()
                    .alpha(if (canContinue) 1f else 0.5f)
            ) {
                Text(
                    text = context.getString(R.string.action_continue),
                    fontSize = 18.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = Color.White,
                    modifier = Modifier.padding(vertical = 4.dp)
                )
            }
        }
    }
}
